### ISSUE TITLE
model with a hasMany without a belongsTo save() PATCH does not persist

### ISSUE THREAD

**Issue by jrburke** [samselikoff/ember-cli-mirage#904]
- Ember 2.8.x with ember-cli-mirage 0.2.2
- [Example project](https://github.com/jrburke/mirage-hasmany)

[models/library.js](https://github.com/jrburke/mirage-hasmany/blob/master/app/models/library.js):

``` javascript

export default DS.Model.extend({
    name: attr('string'),
    books: hasMany('book')
});
```

[models/book.js](https://github.com/jrburke/mirage-hasmany/blob/master/app/models/book.js) (does not contain a `belongsTo`):

``` javascript
export default DS.Model.extend({
    name: attr('string')
});
```

When [adding a new book](https://github.com/jrburke/mirage-hasmany/blob/master/app/controllers/counties/library/book/new.js):

``` javascript
const book = this.get('store').createRecord('book', {
    name,
});

book.save().then(() => {
    library.get('books').pushObject(book);
    return library.save();
}).then(() => {
    this.transitionToRoute('counties.library', library);
});
```

The PATCH request to mirage for the `library` does not seem to save the `book` relationship to account for the new `book` that was added, even though the PATCH request does contain the new book in the relationship.

The [mirage config](https://github.com/jrburke/mirage-hasmany/blob/master/mirage/config.js) just has a simple `this.patch('/libraries/:id');` entry in it.

I am new to ember-cli-mirage and ember data so I am likely just missing something pretty basic.

Questions:

1) How would I change things so that the new book is associated with the `library`?

2) Is there a better way to do this that just results in one call to the server to create the `book` and update the relationship to the `library`?

For other business reasons, adding a `belongsTo('library')` to the `book` is not desirable since we could have `books` that may not be associated with a `library`, or may be associated with a `bookstore` instead of a `library`.

**Comment by samselikoff** [comment]
First question is, how will your actual server persist this relationship? What API format are you using?

Typically the way to persist this all in one request is to have the `belongsTo` side on the book as well. Note that you can add `belongsTo: library()`, even if that relationship is sometimes null (i.e. the book may not be associated with a library). This would add the `libraryId` foreign key to the book, and let you create a new associated book with a single request.

**Comment by blipntime** [comment]
This does not work for me. I am unable to create an association with the code above.

**Comment by huberts** [comment]
I must confirm - I have a model with a has-many relationship to a dictionary (e.g. Person has-many Privileges) - and after patching the Person with new Privileges, the returning object visible via console shows no changes.
I think it should be discouraged to define reflexive belongs-to relationship beetween Privilege and Person.

**Comment by samselikoff** [comment]
One-way associations is the solution to this problem and support landed in 0.3. Please read [the blog post](http://www.ember-cli-mirage.com/blog/2017/01/09/0-3-0-beta-series/) for more information including examples.

[Here is the example of a one-way hasMany](https://ember-twiddle.com/802cd2f92e8c6c7280f42b054f393097?openFiles=models.book.js%2C), which is similar to the OP's problem.
